scripts/bwctest_osd.sh

```diff
@@ -171,115 +164,60 @@ test_suites=(
   ["osd-1.1.0"]=$TEST_GROUP_4
 )
 
-# remove the running opensearch process
-function clean {
-  echo "Closing the running OpenSearch"
-  process=($(ps -ef | grep "Dopensearch" | awk '{print $2}'))
-  kill ${process[0]}
-  echo "Closing any usage on port $BIND_PORT"
-  process=($(lsof -i -P -n | grep $BIND_PORT | awk '{print $2}'))
-  kill ${process[0]}
-}
-
-# Print out a textfile line by line
-function print_txt {
-  while IFS= read -r line; do
-    echo "text read from $1: $line"
-  done < $1
-}
-
-# this function is used to check the running status of OpenSearch or OpenSearch Dashboards
-# $1 is the path to the tmp file which saves the running status 
-# $2 is the error msg to check
-# $3 is the url to curl
-# $4 contains arguments that need to be passed to the curl command
-function check_status {
-  while [ ! -f $1 ] || ! grep -q "$2" $1; do 
-     if [ -f $1 ]; then rm $1; fi  
-     curl $3 $4 > $1 || true
-  done
-  rm $1
-}
-
 # this function sets up the cypress env
 # it first clones the opensearch-dashboards-functional-test library
 # then it removes the tests into the cypress integration folder 
 # and copies the backwards compatibility tests into the folder
-function setup_cypress {
+function setup_cypress() {
   echo "[ Setup the cypress test environment ]"
-  git clone https://github.com/opensearch-project/opensearch-dashboards-functional-test "$test_dir"
-  rm -rf "$test_dir/cypress/integration"
-  cp -r "$cwd/cypress/integration" "$test_dir/cypress"
-  cd "$test_dir"
+  git clone --depth=1 https://github.com/opensearch-project/opensearch-dashboards-functional-test "$TEST_DIR"
+  rm -rf "$TEST_DIR/cypress/integration"
+  cp -r "$CWD/cypress/integration" "$TEST_DIR/cypress"
+  [ ! -d "$LOGS_DIR" ] && mkdir -p "$LOGS_DIR"
+  cd "$TEST_DIR"
   npm install
   echo "Cypress is ready!"
 }
 
-# this function copies the tested data for the required version to the opensearch data folder
-# $1 is the required version
-function upload_data {
-  rm -rf "$opensearch_dir/data"
-  cd $opensearch_dir
-  cp "$cwd/cypress/test-data/$dashboards_type/$1.tar.gz" . 
-  tar -xvf "$opensearch_dir/$1.tar.gz" >> /dev/null 2>&1
-  rm "$1.tar.gz"
-  echo "Data has been uploaded and ready to test"
-}
-
-# Starts OpenSearch, if verifying a distribution it will install the certs then start.
-function run_opensearch {
-  echo "[ Attempting to start OpenSearch... ]"
-  cd "$opensearch_dir"
-  [ $SECURITY_ENABLED == "false" ] && ./bin/opensearch || ./opensearch-tar-install.sh
+function run_cypress() {
+    [ $1 == "core" ] && IS_CORE=true || IS_CORE=false
+    TEST_ARRAY=("$@")
```

Comment:
Can we call this `tests`, `test_names`, or something else to eliminate the duplication with the global array for test versions (`test_array`)?

Reply:
I went ahead and left this if that's cool I updated the other to be TEST_VERSIONS.